internal/rpcserver/rpcserver.go

```diff
@@ -2748,16 +2800,40 @@ func handleGetRawTransaction(_ context.Context, s *Server, cmd interface{}) (int
 	var blkHeight int64
 	var blkIndex uint32
 	chain := s.cfg.Chain
+	txIndex := s.cfg.TxIndexer
 	tx, err := s.cfg.TxMempooler.FetchTransaction(txHash)
 	if err != nil {
-		if s.cfg.TxIndexer == nil {
+		if txIndex == nil {
 			return nil, rpcInternalError("The transaction index "+
 				"must be enabled to query the blockchain "+
 				"(specify --txindex)", "Configuration")
 		}
 
+		// Ensure the tx index is synced.
+		tHeight, tHash, err := txIndex.Tip()
+		if err != nil {
+			return nil, rpcInternalError(err.Error(), "Tip")
+		}
+
+		chain := s.cfg.Chain
+		if chain.BestSnapshot().Height > (tHeight + 5) {
```

Comment:
We may want to add a comment here and/or define `5` as a constant with a description.